### PR TITLE
Update kotlin to 1.4.10

### PR DESCRIPTION
**Pull request by planarvoid**
This PR updates Kotlin to 1.4.10, Coroutines to 1.3.9 and Serialization to 1.0-M1-1.4.0-rc. 

Related FluxC PR - https://github.com/wordpress-mobile/WordPress-FluxC-Android/pull/1688

I had to also update the serialization library because without this change I was getting the following build error:
```
e: java.lang.NullPointerException
        at org.jetbrains.kotlinx.serialization.compiler.resolve.SearchUtilsKt.getKSerializerConstructorMarker(SearchUtils.kt:65)
        at org.jetbrains.kotlinx.serialization.compiler.resolve.KSerializerDescriptorResolver.createLoadConstructorDescriptor(KSerializerDescriptorResolver.kt:302)
        at org.jetbrains.kotlinx.serialization.compiler.extensions.SerializationResolveExtension.generateSyntheticSecondaryConstructors(SerializationResolveExtension.kt:90)
        at org.jetbrains.kotlin.resolve.extensions.SyntheticResolveExtension$Companion$getInstance$1.generateSyntheticSecondaryConstructors(SyntheticResolveExtension.kt:129)
        at org.jetbrains.kotlin.resolve.lazy.descriptors.LazyClassMemberScope.addSyntheticSecondaryConstructors(LazyClassMemberScope.kt:472)
```

Do you think you could check whether the changes I made to the Stories project make sense @mzorz? Update of the serialization library resulted in some changes to the code there. 

I think once we approve this PR, we have to release updated versions of all the modules. 

To test:
- Test that the app behaves as expected - there should be no functional changes
- Check that the tests pass

PR submission checklist:

- [x] I have considered adding unit tests where possible.
- [x] I have considered adding accessibility improvements for my changes.
- [x] I have considered if this change warrants user-facing release notes and have added them to `RELEASE-NOTES.txt` if necessary.

### CODE REVIEW

**Review by mzorz — APPROVED** [review]
Tested the stories library integration part through https://github.com/Automattic/stories-android/pull/540#pullrequestreview-524063679
Approving :shipit: will keep this unmerged though, waiting for @malinajirka's final review

**Review by malinajirka — APPROVED** [review]
LGTM, thanks! ;)

**Pull request merged**
